pypit/arsciexp.py

```diff
@@ -466,6 +471,88 @@ def MasterFlatField(self, fitsdict, det):
         self.SetMasterFrame(mspixelflatnrm, "normpixelflat", det)
         return True
 
+    def MasterSlitFlat(self, fitsdict, det):
+        """
+        Generate Master slit flat frame for a given detector
+
+        Parameters
+        ----------
+        fitsdict : dict
+          Contains relevant information from fits header files
+        det : int
+          Index of the detector
+
+        Returns
+        -------
+        boolean : bool
+          Should other ScienceExposure classes be updated?
+        """
+        dnum = settings.get_dnum(det)
+        # If the master edge is already made, use it
+        if self._msslitflat[det - 1] is not None:
+            msgs.info("An identical master slit frame already exists")
+            return False
+        if settings.argflag['reduce']['slitedge']['useframe'] in ['trace', 'slitflat']:
```

Comment:
I agree that we should include a description in the docs about the various types of flat frames - and I'll include this before we merge. Before I do so, we should discuss the naming for the various flat types.

For Levy, it is necessary to use a pinhole ("trace") flat to first trace the centre of the slit, because the adjacent slit edges are too close to the science slit. A flatfield frame taken with the same slit length as the science frame is then used to trace the slit edges ("slitflat"). I think you may have previously referred to this as an edgeflat? I agree that this naming is a little confusing, so what is your preference? These are the options that I can think of (feel free to add more!).

pixelflat - pixel to pixel variations
slitflat/edgeflat/traceflat - trace slit edges (and determine the echelle blaze)
traceflat/pinhole/centroid - trace the slit centre